scality_manila_utils/export.py

```diff
@@ -0,0 +1,250 @@
+
+    def __ne__(self, other):
+        eq = self == other
+
+        if isinstance(eq, bool):
+            return not eq
+        else:
+            return NotImplemented
+
+
+class Export(object):
+    """
+    Represents an exported filesystem, i.e. a single line in /etc/exports.
+    """
+    __slots__ = ('export_point', 'clients')
+
+    # Naive pattern, matching anything similar to an ip, hostname with wildcard
+    # combinations followed by optional mount options
+    CLIENT_PATTERN = re.compile(
+        r'^(?P<host>[a-z0-9*.-/]+)([(](?P<options>.+)[)])?$'
+    )
+
+    def __init__(self, export_point, clients):
+        """
+        :param export_point: the export point or filesystem
+        :type export_point: string
+        :param clients: a mapping from hostname/network to a set of export
+            options
+        :type clients: dict
+        """
+        # Exported filesystem
+        self.export_point = export_point
+
+        # Hosts, networks allowed to mount the filesystem and corresponding
+        # export options
+        if not clients:
+            raise ExportException('An export must have at least one client')
+        self.clients = clients
+
+    @classmethod
+    def deserialize(cls, line):
+        """
+        Deserialize a line of an /etc/exports file.
+
+        A line contains an export point and a whitespace-separated list of
+        clients allowed to mount the file system at that point;
+            <export> <host1>(<options>) <host2>(<options>)...
+
+        :param line: a line in /etc/exports format
+        :type line: string
+        :returns: :py:class:`scality_manila_utils.export.Export` instance
+        """
+        def client_extract(client):
+            match = cls.CLIENT_PATTERN.match(client)
+            if match is None:
+                msg = "Unable to parse client from {0:s}".format(client)
+                raise DeserializationException(msg)
+
+            host = match.group('host')
+            option_list = match.group('options')
+            if option_list is None:
+                options = set()
+            else:
+                options = set(option_list.split(','))
+            return host, options
+
+        export_parts = line.split()
+
+        if len(export_parts) < 2:
+            msg = "'{0:s}' is not a valid export line".format(line)
+            raise DeserializationException(msg)
+
+        export_point = export_parts[0]
+        clients = dict(map(client_extract, export_parts[1:]))
+        return Export(export_point, clients)
+
+    def serialize(self):
+        """
+        Serialize this `Export` to an /etc/exports representation.
+
+        :returns: a string in /etc/exports format
+        """
+        clients = ''
+        for host, options in self.clients.items():
+            clients += ' ' + host
+            if options:
+                clients += '({0:s})'.format(','.join(options))
+
+        export_line = '{export_point:<32s} {clients:s}'.format(
```

Comment:
This is just an attempt to have some alignment of the clients in the exports file. There is nothing more to it.